src/materials/GeneralizedExternalDrivingForce.C

```diff
@@ -0,0 +1,88 @@
+//* This file is part of the RACCOON application
+//* being developed at Dolbow lab at Duke University
+//* http://dolbow.pratt.duke.edu
+///
+// Aditya Kumar, Blaise Bourdin, Gilles A. Francfort, Oscar Lopez-Pamies,
+// Revisiting nucleation in the phase-field approach to brittle fracture,
+// Journal of the Mechanics and Physics of Solids,
+// Volume 142,
+// 2020,
+// 104027,
+// ISSN 0022-5096,
+// https://doi.org/10.1016/j.jmps.2020.104027.
+// (https://www.sciencedirect.com/science/article/pii/S0022509620302623)
+
+#include "Function.h"
+#include "GeneralizedExternalDrivingForce.h"
+
+registerADMooseObject("raccoonApp", GeneralizedExternalDrivingForce);
+
+InputParameters GeneralizedExternalDrivingForce::validParams() {
+  InputParameters params = Material::validParams();
+  params.addClassDescription("computes the Kumar coeffs"
+                             "Ce()");
+  params.addParam<MaterialPropertyName>(
+      "energy_release_rate", "Gc", "energy release rate or fracture toughness");
+  params.addParam<MaterialPropertyName>(
+      "phase_field_regularization_length", "l",
+      "the phase field regularization length");
+  params.addParam<MaterialPropertyName>("Lambda", "Lambda",
+                                        "Lame's first parameter Lambda");
+  params.addParam<MaterialPropertyName>("shear_modulus", "G",
+                                        "shear modulus mu or G");
+  params.addRequiredParam<Real>("tensile_strength",
+                                "critical tensile strength");
+  params.addRequiredParam<Real>("compressive_strength",
+                                "critical compressive strength");
+  params.addRequiredParam<Real>("delta", "delta");
+  params.addParam<MaterialPropertyName>("rank_two_tensor", "stress",
```

Comment:
Don't call it `rank_two_tensor`. I think it is better to call this parameter `stress`:
```suggestion
  params.addParam<MaterialPropertyName>("stress", "stress",
```